### PR TITLE
fix: build process causing esm issues

### PR DESCRIPTION
**Pull request by kylemcd**

### CODE REVIEW

File: packages/react/vite.config.ts [hunk 1]
```diff
@@ -61,8 +69,7 @@ export default defineConfig(({ mode }) => {
 
                 if (file?.type === "chunk") {
                   // Replace .css imports and requires
-                  const pattern =
```
Comment (connorlindsey) [inline packages/react/vite.config.ts:64]: We need to keep the `require` portion of the regex as that strips the unnecessary `.css.js` imports from the cjs build. For example, if you build and look at `react/dist/cjs/modules/core/components/Button/Button.js` you'll see it includes `require("./styles.css.js");`

`/(import\s+['"].+\.css\..*?['"];?)|(require\(['"][^()]+\.css(\.js)?['"]\);?)/g`

File: packages/react/vite.config.ts [hunk 2]
```diff
@@ -9,34 +13,35 @@ import execute from "rollup-plugin-execute";
...
         outDir: "dist/types",
       }),
       react(),
-      noBundlePlugin({ root: "./src" }),
+      // splitVendorChunkPlugin(),
+      noBundlePlugin({ root: resolve(__dirname, "src") }),
     ],
     build: {
-      outDir: CJS ? "dist/cjs" : "dist/esm",
+      // cssMinify: "lightningcss",
```
Comment (connorlindsey) [inline packages/react/vite.config.ts:28]: Do we want to change css minifier? I didn't look too closely at the different options.

File: packages/react/vite.config.ts [hunk 3]
```diff
@@ -50,7 +55,10 @@ export default defineConfig(({ mode }) => {
             // Move index.css to root of dist
             `mv dist/esm/index.css dist/index.css`,
             // Delete extra .css.js files
-            `rm dist/**/*.css.*`,
+            `find ./dist -name "*.css.js" -delete`,
```
Comment (connorlindsey) [inline packages/react/vite.config.ts:58]: Nit: I think we could get this to a single line catching anything `.css.*`: ``find ./dist -name "*.css.*" -delete``

**Review by connorlindsey — COMMENTED** [review]
Tested this against Next 12/13/14, Blitz, and CRA and I think this is a good fix for the ESM issues we were seeing, just a bit of clean up around comments/imports

File: packages/react/vite.config.ts (hunk 2 above)
Comment (kylemcd) [reply packages/react/vite.config.ts:28]: Ah I was testing something out, isn't a big deal here, will just remove the commented code.

File: packages/react/package.json [hunk 4]
```diff
@@ -68,6 +69,7 @@
     "eslint-plugin-react-hooks": "^4.6.0",
     "eslint-plugin-react-refresh": "^0.4.4",
     "jsdom": "^24.0.0",
+    "lightningcss": "^1.23.0",
```
Comment (connorlindsey) [inline packages/react/package.json:72]: If we're not using lightningcss, let's remove the dependency and re-install to update the lock file

File: packages/react/vite.config.ts [hunk 5]
```diff
@@ -9,34 +9,33 @@ import execute from "rollup-plugin-execute";
...
-        // External packages that should not be bundled
-        external: ["react", "react-dom"],
+        external: ["react"],
         output: {
           interop: "compat",
-          format: formats[0],
           globals: {
             react: "React",
           },
+          // External packages that should not be bundled
```
Comment (connorlindsey) [inline packages/react/vite.config.ts:38]: Nit: Comment should be above the `external` option

**Review by connorlindsey — APPROVED** [review]
Little bit more clean up but approving to not block getting this out

**Pull request merged**